src/services/submissionService.ts

```diff
@@ -79,14 +114,17 @@ export const addSubmission = async (submission: gqlType.Submission):
     try {
         const dbSubmission = convertToDbSubmission(submission)
 
-        const db = getFirestore()
-        const submissionRef = db.collection('submissions')
+        const submissionRef = dbInstance.collection('submissions')
 
         const docRef = await submissionRef.add(dbSubmission)
 
+        const currentDate = new Date().toISOString()
+
         const newSubmission: dbSchema.Submission = {
             ...dbSubmission,
-            id: docRef.id
+            id: docRef.id,
+            createdDate: currentDate,
```

Comment:
Nit: we should move the logic to set the date here so it can't accidentally be passed in.

Reply:
@ermish I think I've fixed this. When I pass in the `createdDate` like this:
```
{
  "input": {
    "createdDate": "2023-09-11T02:01:09.298Z"
    "googleMapsUrl": "https://maps.google.com/exampleA",
    "healthcareProfessionalName": "Dr X",
    "spokenLanguages": [
      {
        "iso639_3": "eng",
        "nameJa": "English",
        "nameEn": "English",
        "nameNative": "English"
      }
    ]
  }
}
```
I get the following error:
```
"code": "BAD_USER_INPUT",
        "stacktrace": [
          "GraphQLError: Field \"createdDate\" is not defined by type \"SubmissionInput\".",
```
Let me know if this is different from what you had in mind!

Reply:
sorry, I put this comment on the wrong part. my mistake. no need to add it here. The ideal spot is before it gets added to the db which you have the logic now in the conversion method. so this can be removed.